### ISSUE TITLE
Gradient checkpointing issue when running QLoRA finetuning

### ISSUE THREAD

**Issue by tytung2020** [mosaicml/llm-foundry#413]
Finetuning the mpt-7b and mpt-30b using qlora gives the above mentioned error. Is there a way to fix this?

**Comment by tytung2020** [comment]
are these lines of codes what is needed to make it work? cekal's amendment seems to work on the 7b version:
https://huggingface.co/cekal/mpt-7b-peft-compatible/commit/a5eab52c1c61c1d50a4e01428949f6ff90c73c48
But not sure if it works fully as intended. Could someone in MosaicML check this? Thanks
